app/src/main/java/com/google/samples/apps/sunflower/compose/plantdetail/PlantDetailView.kt

```diff
@@ -0,0 +1,421 @@
+    transitionModifier: Modifier = Modifier
+) {
+    var imageHeight by state { 0 }
+
+    Stack(modifier.fillMaxWidth()) {
+        PlantImage(scrollerPosition, imageUrl, transitionModifier.onPositioned {
+            imageHeight = it.size.height
+        })
+        if (!isPlanted) {
+            val fabModifier = if (imageHeight != 0) {
+                Modifier
+                    .gravity(Alignment.TopEnd)
+                    .padding(end = 8.dp)
+                    .offset(y = getFabOffset(imageHeight, scrollerPosition))
+                    .plus(transitionModifier)
+            } else {
+                Modifier
+            }
+            val fabAccessibilityLabel = stringResource(R.string.add_plant)
+            FloatingActionButton(
+                onClick = onFabClicked,
+                shape = MaterialTheme.shapes.small,
+                modifier = fabModifier.semantics {
+                    accessibilityLabel = fabAccessibilityLabel
+                }
+            ) {
+                Icon(Icons.Filled.Add)
+            }
+        }
+    }
+}
+
+@Composable
+private fun PlantImage(
+    scrollerPosition: ScrollerPosition,
+    imageUrl: String,
+    modifier: Modifier = Modifier,
+    placeholderColor: Color = MaterialTheme.colors.onSurface.copy(0.2f)
+) {
+    val parallaxOffset = with(DensityAmbient.current) {
+        scrollerParallaxOffset(scrollerPosition)
+    }
+    CoilImageWithCrossfade(
+        data = imageUrl,
+        contentScale = ContentScale.Crop,
+        loading = {
+            Box(modifier = Modifier.fillMaxSize(), backgroundColor = placeholderColor)
+        },
+        modifier = modifier.fillMaxWidth().padding(top = parallaxOffset).preferredHeight(278.dp)
+    )
+}
+
+@Composable
+private fun PlantHeaderActions(
+    onBackClicked: () -> Unit,
+    onShareClicked: () -> Unit,
+    modifier: Modifier = Modifier
+) {
+    Row(
+        modifier = modifier.fillMaxSize().systemBarsPadding(top = true).padding(top = 12.dp),
+        horizontalArrangement = Arrangement.SpaceBetween
+    ) {
+        val iconModifier = Modifier.sizeIn(maxWidth = 32.dp, maxHeight = 32.dp)
+            .drawBackground(color = MaterialTheme.colors.surface, shape = CircleShape)
+
+        IconButton(
+            onClick = onBackClicked,
+            modifier = Modifier.padding(start = 12.dp).plus(iconModifier)
+        ) {
+            Icon(Icons.Filled.ArrowBack)
+        }
+        val shareAccessibilityLabel = stringResource(R.string.menu_item_share_plant)
+        IconButton(
+            onClick = onShareClicked,
+            modifier = Modifier.padding(end = 12.dp).plus(iconModifier).semantics {
+                accessibilityLabel = shareAccessibilityLabel
+            }
+        ) {
+            Icon(Icons.Filled.Share)
+        }
+    }
+}
+
+@Composable
+private fun PlantInformation(
+    name: String,
+    wateringInterval: Int,
+    description: String,
+    onNamePositioned: (LayoutCoordinates) -> Unit,
```

Comment:
Does this need to send the entire `LayoutCoordinates`? Could it be narrowed?

Reply:
Thanks!